samples/remote_attestation_eeid/common/attestation.cpp

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include "attestation.h"
+#include <string.h>
+#include "log.h"
+
+Attestation::Attestation(Crypto* crypto, uint8_t* enclave_mrsigner)
+{
+    m_crypto = crypto;
+    m_enclave_mrsigner = enclave_mrsigner;
+}
+
+/**
+ * Generate a remote report for the given data. The SHA256 digest of the data is
+ * stored in the report_data field of the generated remote report.
+ */
+bool Attestation::generate_remote_report(
+    const uint8_t* data,
+    const size_t data_size,
+    uint8_t** remote_report_buf,
+    size_t* remote_report_buf_size)
+{
+    bool ret = false;
+    uint8_t sha256[32];
+    oe_result_t result = OE_OK;
+    uint8_t* temp_buf = NULL;
+
+    if (m_crypto->Sha256(data, data_size, sha256) != 0)
+    {
+        goto exit;
+    }
+
+    // To generate a remote report that can be attested remotely by an enclave
+    // running  on a different platform, pass the
+    // OE_REPORT_FLAGS_REMOTE_ATTESTATION option. This uses the trusted
+    // quoting enclave to generate the report based on this enclave's local
+    // report.
+    // To generate a remote report that just needs to be attested by another
+    // enclave running on the same platform, pass 0 instead. This uses the
+    // EREPORT instruction to generate this enclave's local report.
+    // Both kinds of reports can be verified using the oe_verify_report
+    // function.
+    result = oe_get_report(
+        OE_REPORT_FLAGS_REMOTE_ATTESTATION,
+        sha256, // Store sha256 in report_data field
+        sizeof(sha256),
+        NULL, // opt_params must be null
+        0,
+        &temp_buf,
+        remote_report_buf_size);
+    if (result != OE_OK)
+    {
+        TRACE_ENCLAVE("oe_get_report failed.");
+        goto exit;
+    }
+    *remote_report_buf = temp_buf;
+    ret = true;
+    TRACE_ENCLAVE("generate_remote_report succeeded.");
+exit:
+    return ret;
+}
+
+/**
+ * Attest the given remote report and accompanying data. It consists of the
+ * following three steps:
+ *
+ * 1) The remote report is first attested using the oe_verify_report API. This
+ * ensures the authenticity of the enclave that generated the remote report.
+ * 2) Next, to establish trust of the enclave that  generated the remote report,
+ * the mrsigner, product_id, isvsvn values are checked to  see if they are
+ * predefined trusted values.
+ * 3) Once the enclave's trust has been established, the validity of
+ * accompanying data is ensured by comparing its SHA256 digest against the
+ * report_data field.
+ */
+bool Attestation::attest_remote_report(
```

Comment:
We shouldn't provide EEID as an additional parameter. A better approach is to utilize the attestation plugin model, and turn EEID as one of the claims to be consumed by the verifier.

Reply:
Absolutely! I haven't ported this example yet, it still uses the older interface from before we had the plugin. Also, I'm not sure we want to keep this as a separate example.

Reply:
I think a sample would be nice to have. We just need to implement it with the plugin model, and have the attester and verifier in different host applications.